index.js

```diff
@@ -1228,50 +1228,29 @@ RedisClient.prototype.eval = RedisClient.prototype.EVAL = function () {
     });
 };
 
+exports.createClient = function(arg0, arg1, options) {
+    if (typeof arg0 === 'object' || arg0 === undefined) {
+        options = arg0 || options;
+        return createClient_tcp(default_port, default_host, options);
+    }
+    if (typeof arg0 === 'number' || typeof arg0 === 'string' && arg0.match(/^\d+$/)){
+        return createClient_tcp(arg0, arg1, options);
+    }
+    if (typeof arg0 === 'string') {
+        options = arg1 || {};
 
-exports.createClient = function(arg0, arg1, arg2){
-    if( arguments.length === 0 ){
-
-        // createClient()
-        return createClient_tcp(default_port, default_host, {});
-
-    } else if( typeof arg0 === 'number' ||
-        typeof arg0 === 'string' && arg0.match(/^\d+$/) ){
-
-        // createClient( 3000, host, options)
-        // createClient('3000', host, options)
-        return createClient_tcp(arg0, arg1, arg2);
-
-    } else if( typeof arg0 === 'string' ){
-        var parsed = URL.parse(arg0, true, true),
-            options = (arg1 || {});
-
+        var parsed = URL.parse(arg0, true, true);
```

Comment:
It looks like `URL.parse` does not return a `parsed.port` when the string is only a string number like "6379", so this introduces breaking changes. Am I wrong?

```
erin@elisheva:~/node_redis$ node  -e "var URL = require('url'); console.log(URL.parse('8000'));"
{ protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: '8000',
  path: '8000',
  href: '8000' }
```